### ISSUE TITLE
Nuget removes analyzers from projects when updating.

### ISSUE THREAD

**Issue by JohanLarsson** [NuGet/Home#6035]
Please read the following information before posting the issue.

## Details about Problem

NuGet VS UI

NuGet latest?

VS 2017

OS win10

## Detailed repro steps so we can see the same problem

1. Clone https://github.com/nestquik/diploma/commit/a7d899bc36910723a5cc1d7401f319331ea6f4ba

2. Install https://www.nuget.org/packages/PropertyChangedAnalyzers/2.0.0.1-dev

Check git diff, nothing added to projects.
Downgrade to 1.0.0 of the analyzer, works.
Upgrade to 2.0.01, removes the analyzer.

**Comment by sharwell** [comment]
:memo: When I reduced this problem locally, I performed the following steps:

1. Install PropertyChangedAnalyzers version 1.0.0
2. Update to PropertyChangedAnalyzers 2.0.0.1-dev

When following these steps, the project file was correct after step 1 but incorrect after step 2. It seemed the project file was correct after every other action, so if you don't reproduce it on the first try, perform an additional step and check again.

**Comment by jainaashish** [comment]
I don't know what NuGet can do here... install scripts is being executed with each install/update action, so there might be something going wrong with Project System.

**Comment by JohanLarsson** [comment]
What is wrong with the scripts?

**Comment by rrelyea** [comment]
@JohanLarsson / @sharwell  - Does this reproduce with any other analyzers?

@jainaashish says that it also reproduced updating from a released version to another released version. He was using 15.5 preview2

**Comment by JohanLarsson** [comment]
@rrelyea I don't use nuget much so don't know much about how it works with other analyzers. It installed fine using paket if that is worth anything.

I'm not sure the issue is the prerelease, that was just my first guess.

**Comment by duncanp-sonar** [comment]
This issue repros with the SonarAnalyzer.CSharp analyzer which is not marked as pre-release. It will also repro if you create a new Roslyn analyzer from scratch in VS2017 using the _Analyzer with Code Fix (NuGet + VSIX) template_.

Some further investigation:
* the same analyzer NuGet packages update correctly in VS2015.3 for .Net Framework projects.
* they update correctly for .Net Core projects in VS2017.4.1
* but they do not update correctly .Net Framework projects in VS2017.4.1.

Non-analyzer NuGet packages seem to be ok in both versions of VS and for both project types.

**Comment by JohanLarsson** [comment]
Here is a repro using https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers/
[ClassLibrary1.zip](https://github.com/NuGet/Home/files/1427630/ClassLibrary1.zip)

Steps to reproduce:
1) Update-Package Microsoft.CodeAnalysis.Analyzers
2) Check the project file.
The rows:
```xml
    <Analyzer Include="..\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.Analyzers.dll" />
    <Analyzer Include="..\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll" />
```
Are deleted and not replaced with rows pointing to the updated package.
`packages.config` is updated.

@rrelyea I have 100% repro updating all analyzers I have tried.

**Comment by xt0rted** [comment]
I'm seeing this with [FakeItEasy.Analyzer.CSharp](https://www.nuget.org/packages/FakeItEasy.Analyzer.CSharp/). In a brand new project these install just fine initially, but when you update them every other update will remove the reference to the analyzer. It's pretty annoying but now that I know that's what happens I can downgrade, then upgrade to the newest version without it being removed from the project.

**VS Version:** 15.4.3
**NuGet Version:** 4.4.0
**Project type:** classic csproj
**.NET:** 4.6.2 & 4.7

A secondary issue I've noticed is in VS 15.5.0 Preview 3 w/ NuGet 4.5.0. When I tried to upgrade from 4.0.0 to 4.1.1 of that analyzer it added a reference to the new version, but didn't remove the old reference leaving me with two references. The same issues exist where every other install removes the reference and doesn't install the new one.

**Comment by jainaashish** [comment]
I investigated this and the problem is NuGet isn't using the updated dte project to run install.ps1 which add these analyzers so when you update, it still uses old dte project reference which still has these analyzers so install.ps1 skips adding these. Next time, since there were no analyzers so it add them and this continues forever...

Thanks everyone for pointing this, we'll get this fix with upcoming release.

**Comment by jainaashish** [comment]
Reopening to track it for 15.5

**Comment by jainaashish** [comment]
Merged in release-4.5.0 branch with 4ea0fb23ddc1fa853e5c3e1b3d81fc7c97cd19c6
